### ISSUE TITLE
[Feature] graph highlight both nodes and edges

### ISSUE THREAD

**Issue by hx425072624** [apache/echarts#19368]
### What problem does this feature solve?

I want to highlight the nodes and edges like the picture below by using dispatchAction,but the function can only highlight nodes or edges.Is there any way to achieve this target
<img width="280" alt="截屏2023-12-07 13 51 06" src="https://github.com/apache/echarts/assets/23489383/9af0888b-1036-41eb-b82c-03c6267b969c">

### What does the proposed API look like?

dispatchAction({
type:'highlight',
batch:[
{
dataIndex:[1,2,34,5,6],
dataType:'node'
}
{
dataIndex:[1,2,34,5],
dataType:'edge'
}
]
})

**Comment by helgasoft** [comment]
> using dispatchAction, but the function can only highlight nodes or edges.

Nope, it can do **both** simultaneously - [Demo Code](https://echarts.apache.org/examples/en/editor.html?c=graph-simple&code=PYBwLglsB2AEC8sDeAoWtJgDYFMBcya6GOAHmAQOQBCAhgM4QDGsA4gE60gAWlRAvgBoiYYMCyQQBJENhFa0CAFtakGHgBmtLPRzD0C5aqjQAIgFdOa6AFUQAE1X5YARgCsABg_7YhldYBRBghoAHM7RzBnSgBHcxDIJgBJaAB5czBKH112CBx6AgBtInRUYmIwAE8QaNDOHiyS4ixaSuAMqmgYHEby9HpKpQAjcQBlCAAvZ08fcvZgWiUCMHZzPSacJR5ggqQIKKVRqtxpJnFgdjxKdhx7SiEmvr6sEJwjypPkM6wLq5u7_gPJ4tIY4LDSWD0bjAADuy1WOFgQL6t1Cb0GI3BsEKlCYEHYTFwWVglFo7HmMMoAF1ZsRUejhmNJs5CgAWQSuDw0x6wekAGVooKxZSesA0MDA4ymBAATB4ecjypFaEUeaU1cRoItogA5YD2REuXqi9CkAgAZi8tKelQtXg1ir6IpNWqUuv1iJlxpNZtgAA4rRr0LbYJb5SbHeVnaLXe6DaHvaLfW4ZkHYCGXPaI9biNGnrGqHr46zE09k6mTcGCJ5w6L-DzuU8APRN2AvaAAawK2MbzxCXdVJrzfXo7QJzm8abAZLRFFcOfKA0ZWClLLcHLlvZNILB0jT_WhcLF2l0acjfegb2OzmHTxhEHsYG41YXfSYlgAbjhL_Rux4AHQyq-5SbNsjC7N8vzXLc9xnq-57qpWkJjkwcaeqWfTTuws6Fh6rgYeUO7CvukKHvCaxwWm7ZXh8N4ke-7Bfj-f6AWeDqvrexCjpYqG4fGRrARgM44HOlBFoi5p8NmGqcf0KFobAXqCVhOEkuJCbsTJabceOfHocpwmiepJaaUO2nyXp-EGdhImWSZEYNjm1HvJ8nGgLQeJVAQAEAJyvvej7Popr4MUx-R_gqTT1ugVIoPwADcKBKJUADC3BkmA_4wAAFLiLxMB2xIaOY0BMNY2U4B-ACUIrJWlGX_vYED0CAqhMNwACCZUmNlSByOUVQ1Fc9iwtAIAtJUxJNEMbXPoUSA5Hk9ApAavoeEiHJIMqAAq1TRPS9yxeg_BVYl6B1el7CZU1LWzV15VNH1g3RNwEChNwLzvZkHKLfkK1kHg3j9cQM1gO1eDFE8W2qLQ_2-tilWNTDcMcojypwwA1C4sBUue0PTrtQ0kgdHLo9Aq1FGjyPk2QqMfkj06Yy4uNNCz6BVXFp1AA).
📌 please close issue if problem solved.

![image](https://github.com/apache/echarts/assets/13038071/034296b3-7ee1-4f9b-8046-eeb03832e07e)

**Comment by hx425072624** [comment]
but this can't fade out of other data to focus the highlighted,is this a bug?

**Comment by helgasoft** [comment]
I doubt _dispatchAction_ can change focus. Focus works only with _mouseover_.

**Comment by hx425072624** [comment]
> I doubt _dispatchAction_ can change focus. Focus works only with _mouseover_.

<img width="877" alt="截屏2023-12-13 11 05 19" src="https://github.com/apache/echarts/assets/23489383/712874f3-b21c-4613-9305-15d314e14342">
But the document shows Focus works with highlight.I want to fade out of other data when i dispatchAction  highlight,but it don't work

**Comment by nanianlisao** [comment]
> But the document shows Focus works with highlight.I want to fade out of other data when i dispatchAction highlight,but it don't work

I had the same problem, and here's my solution

The second time you highlight node or edge, you need to add the `notBlur: true` property and recommend the focus setting `self`.

```
chart?.dispatchAction({
    type: 'highlight',
    batch: [
      { dataType: 'node', dataIndex: nodeIndexes },
      { dataType: 'edge', dataIndex: selected.indexes, notBlur: true },
    ],
  });
```
